rajawali/src/main/java/org/rajawali3d/materials/Material.java

```diff
@@ -1073,7 +1073,12 @@ public void setModelMatrix(Matrix4 modelMatrix) {
         mModelMatrix = modelMatrix;//.getFloatValues();
         mVertexShader.setModelMatrix(mModelMatrix);
 
-        mNormalMatrix.setAll(modelMatrix).setToNormalMatrix();
+        mNormalMatrix.setAll(modelMatrix);
+        try {
+            mNormalMatrix.setToNormalMatrix();
+        } catch (IllegalStateException exception) {
+            // modelMatrix is degenerate (zero scale)
```

Comment:
For something like this, with we prefer an additional commit. The only time we want things rebased would be if the initial PR exceeds a few commits. For example, when I PR the 2.0 changes, there will likely be hundreds of commits. I will rebase this into 1.